ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStorageManager.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.azure.AzureCloudConfig;
+import com.github.ambry.cloud.azure.AzureMetrics;
+import com.github.ambry.cloud.azure.ConnectionStringBasedStorageClient;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Storage manager that tracks {@link AzureStorage} objects.
+ */
+public class AzureStorageManager implements StoreManager {
```

Comment:
I think we need the backup implementation change, but for the higher level class like CloudStorageManager, the code logic should be the same. 
Do we need create this one?

Reply:
Readability and better code.

We need an AzureStorageManager for following reasons.

1. AzureStorageManager creates ABS containers, which CloudStorageManager does not. We could modify it using some configs and if-else statements but as I have mentioned above, readability is an important goal for V2. Meddling with CloudStorageManager creates unreadable code, may break something or may make it harder to review slowing down the overall project. I would just like V1 code to run as it is separate from V2. 

2. CloudStorageManager tracks CloudBlobStore objects in a map<partitionId, CloudBlobStore>. V2 creates AzureStorage objects by implementing the Store interface. I could change the declaration of the map to <partitionId, Store> but as mentioned above, readability and operability are important goals for V2. 

3. CloudStorageManager is sub-optimal. It uses its own locking mechanism instead of using a ConcurrentHashMap, which is what AzureStorageManager does.

Reply:
Why doesn't CloudStorageManager create containers? Or do you mean we are creating different containers between v1 and v2?